src/DI/Compiler.php

```diff
@@ -149,7 +149,15 @@ public function getConfig(): array
 	 */
 	public function setDynamicParameterNames(array $names)
 	{
-		$this->extensions[self::PARAMETERS]->dynamicParams = $names;
+		$extension = $this->extensions[self::PARAMETERS];
+
+		if ($extension instanceof Nette\DI\Extensions\ParametersExtension) {
+			$extension->dynamicParams = $names;
+		} else {
+			throw new InvalidConfigurationException(
+				"Parameter 'dynamicParams' in extension '" . self::PARAMETERS . "' does not exist."
+			);
+		}
```

Comment:
`assert($extension instanceof Nette\DI\Extensions\ParametersExtension)` is better for internal behavior. This exception can never happen and assert is enough to satisfy IDE and also phpstan.

Reply:
does phpstan also understand this? `assert($this->extensions[self::PARAMETERS] instanceof Nette\DI\Extensions\ParametersExtension)`

Reply:
I am not sure. Edge case which I never needed.

Reply:
I tried it and it works.

Reply:
Fixed.